lib/rules/require-valid-css-selector-for-testing-helpers.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+const { getSourceModuleNameForIdentifier } = require('../utils/import');
+const { getPropertyValue, parseCallee, getAncestor } = require('../utils/utils');
+const { isIdentifier } = require('../utils/types');
+const jsdom = require('jsdom'); // eslint-disable-line node/no-unpublished-require
```

Comment:
we'll need to fix this before landing (if we landed now it would break consumers)

were you not able to use sizzle for this?

Reply:
sizzle required a window object to exist in the global scope with a document which it uses to call createElement among other things.

Reply:
Ya, I think jsdom is a bit heavy if a dep to add for this. I took a look at what `stylelint` does for this and it seems like they use the `postcss-selector-parser` which would probably do the trick for us.


```js
const selectorParser = require('postcss-selector-parser');

function isValidSelector(selector) {
  try {
    selectorParser().processSync(selector);
    return true;
	} catch {
    return false;
	}
};
```

What do you think? Ming giving that a whirl?

Reply:
The problem with just parsing the selector is it doesn't do any validation it will parse just about anything.

Reply:
Ok I found a css toolkit which does also validate called [csstree](https://github.com/csstree/csstree), I've updated the PR let me know your thoughts thanks!